borsh/tests/test_de_errors.rs

```diff
@@ -1,11 +1,21 @@
 use borsh::{from_slice, BorshDeserialize};
+use borsh_derive::borsh;
 
+#[borsh(use_discriminant = true)]
 #[derive(BorshDeserialize, Debug)]
```

Comment:
Please, swap these two lines (and wherever applicable in other places you put `borsh` before `derive`). It is easier to read attributes that are relevant to the derives after you named the derives, and it is the norm in Rust ecosystem.